src/pages/Stream/Views/Explore/useLogsFetcher.ts

```diff
@@ -18,7 +19,11 @@ const useLogsFetcher = (props: { schemaLoading: boolean; infoLoading: boolean })
 	const showTable = hasContentLoaded && !hasNoData && !errorMessage;
 
 	useEffect(() => {
-		setLogsStore(setCleanStoreForStreamChange);
+		if (isTimeRangeSetRef.current) {
```

Comment:
Using a ref in this way (to prevent resetting the store on every render except for the initial one) is not a best practice for open-source repositories.